pkg/registries/registries.go

```diff
@@ -104,18 +110,31 @@ func mergedMirrorSets(icspRules []*apioperatorsv1alpha1.ImageContentSourcePolicy
 func EditRegistriesConfig(config *sysregistriesv2.V2RegistriesConf, insecureRegistries, blockedRegistries []string, icspRules []*apioperatorsv1alpha1.ImageContentSourcePolicy) error {
 	// getRegistryEntry returns a pointer to a modifiable Registry object corresponding to scope,
 	// creating it if necessary.
-	// NOTE: We never generate entries with Prefix != Location, so everything in updateRegistriesConfig
-	// only checks Location.
+	// Note: Prefix can be a wildcard entry in the format "*.example.registry.com" so we check the value of
+	// Prefix for a wildcard entry. If Prefix doesn't have a wildcard entry, we check Location for regular entries.
 	// NOTE: The pointer is valid only until the next getRegistryEntry call.
 	getRegistryEntry := func(scope string) *sysregistriesv2.Registry {
 		for i := range config.Registries {
-			if config.Registries[i].Location == scope {
+			reg := config.Registries[i].Location
+			if isValidWildcard(config.Registries[i].Prefix) {
```

Comment:
If both `Location` and `Prefix` are set, a scope should always be found by `Prefix`, wildcard or not.
`reg := ….Prefix; if reg == "" { reg = ….Location }`.

Or maybe, if c/image documents that as supported, just always set `Prefix` only in the code creating entries below, and that would simplify other uses in this function.

Reply:
Fixed.
Always use Prefix if Prefix is set